src/routes.rs

```diff
@@ -58,23 +64,97 @@ pub fn catchers() -> Vec<rocket::Catcher> {
     catchers![not_found]
 }
 
-struct AuthorisedUser(String);
+struct AuthorisedUser {
+    // other useful fields could include organisation home scope permissions
+    username: String,
+    authorized: bool,
+}
+
 impl<'a, 'r> FromRequest<'a, 'r> for AuthorisedUser {
```

Comment:
I think we should move this somewhere else

Reply:
Yeah this is the bit I wasn't happy with. I couldn't work out how to get the data from the request guard to the request, so I used the auth_user like a state machine. username + true = authorised user. "bearer" + true = valid token. Anything else = unauthorized.